metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -894,6 +895,27 @@ def _make_select_columns_for_metrics(
                 else:
                     select_expression = column_reference_expression
 
+                # At this point, the MetricSpec might have the alias in place of the element name, so we need to look
+                # back at where it was defined from to get the metric element name.
+                metric_name = metric_instance.defined_from.metric_name
+                input_measures = self._metric_lookup.measures_for_metric(
+                    metric_reference=MetricReference(metric_name),
+                    column_association_resolver=self._column_association_resolver,
+                )
```

Comment:
Another way to deal with the issue below is to add a different, internal API here that only gives the measures that are direct inputs into the metric. That way, if we add another metric type that takes measures as input anything derived from it won't have anything to do here, and we ensure we only do the coalesce once within a derived metric chain.

Reply:
I like that idea! I'll add that.

Reply:
Added that!